variants/VAKE_F446VE/variant.h

```diff
@@ -0,0 +1,182 @@
+#define PD14  60 // H1_PWM
+#define PD15  61 // H0_PWM
+#define PE0   62 // PS_ON
+#define PE1   63 // E4_STEP
+#define PE2   64 // E4_DIR
+#define PE3   65 // E4_CS
+#define PE4   66 // E3_DIR
+#define PE5   67 // E3_STEP
+#define PE6   68 // E3_CS
+#define PE7   69 // E_STOP
+#define PE8   70 // Z_STOP
+#define PE9   71 // Y_STOP
+#define PE10  72 // X_STOP
+#define PE11  73 // E0_CS
+#define PE12  74 // SPI4_SCK
+#define PE13  75 // SPI4_MISO
+#define PE14  76 // SPI4_MOSI
+#define PE15  77 // Z_STEP
+#define PA0   78 // F1_PWM  or serial
+#define PA1   79 // F2_PWM  or serial
+#define PA4   80 // Z_PROBE_D
+#define PB0   81 // E1_CS
+#define PC1   82 // THERM1_ADC 	ADC
+#define PC0   83 // THERM0_ADC	ADC
+
+// This must be a literal
+#define NUM_DIGITAL_PINS        98
+// This must be a literal with a value less than or equal to to MAX_ANALOG_INPUTS
+#define NUM_ANALOG_INPUTS       6 
+#define NUM_ANALOG_FIRST        84
+
+// On-board LED pin number
+#define LED_BUILTIN             PB14
+#define LED_LD2                 LED_BUILTIN
+#define LED_GREEN               LED_BUILTIN
+
+// On-board user button
+#define USER_BTN                PC10
+
+// Timer Definitions
+// Do not use timer used by PWM pins when possible. See PinMap_PWM.
+#define TIMER_TONE              TIM6
+
+// Do not use basic timer: OC is required
+#define TIMER_SERVO             TIM2  //TODO: advanced-control timers don't work
+
+// UART Definitions
+#define SERIAL_UART_INSTANCE    4 //Connected to ST-Link
+
+// Default pin used for 'Serial' instance (ex: ST-Link)
+// Mandatory for Firmata
+//#define PIN_SERIAL_RX           PA1 //PA10
+//#define PIN_SERIAL_TX           PA0 //PA9
+
+//#define PIN_SERIAL1_RX           PA10 //PA10
+//#define PIN_SERIAL1_TX           PA9 //PA9
+
+#ifdef __cplusplus
+} // extern "C"
+#endif
+/*----------------------------------------------------------------------------
+ *        Arduino objects - C++ only
+ *----------------------------------------------------------------------------*/
+
+#ifdef __cplusplus
+// These serial port names are intended to allow libraries and architecture-neutral
+// sketches to automatically default to the correct port name for a particular type
+// of use.  For example, a GPS module would normally connect to SERIAL_PORT_HARDWARE_OPEN,
+// the first hardware serial port whose RX/TX pins are not dedicated to another use.
+//
+// SERIAL_PORT_MONITOR        Port which normally prints to the Arduino Serial Monitor
+//
+// SERIAL_PORT_USBVIRTUAL     Port which is USB virtual serial
+//
+// SERIAL_PORT_LINUXBRIDGE    Port which connects to a Linux system via Bridge library
+//
+// SERIAL_PORT_HARDWARE       Hardware serial port, physical RX & TX pins.
+//
+// SERIAL_PORT_HARDWARE_OPEN  Hardware serial ports which are open for use.  Their RX & TX
+//                            pins are NOT connected to anything by default.
+#define HAVE_HWSERIAL4
+#define HAVE_HWSERIAL1
+#define SERIAL_PORT_MONITOR     Serial4
```

Comment:
You could use Generic `Serial `instance.
Like this if USB CDC is used then this will also work
```suggestion
#define SERIAL_PORT_MONITOR     Serial
```